x-pack/plugins/enterprise_search/server/routes/app_search/engines.test.ts

```diff
@@ -0,0 +1,177 @@
+  describe('GET /api/app_search/engines', () => {
+    const APP_SEARCH_HOST_CONFIG = 'http://localhost:3002';
+    const AUTH_HEADER = 'Basic 123';
+    const PAGE_INDEX = 1;
+    const TYPE = 'indexed';
+
+    let router: RouterMock;
+    const mockResponseFactory = httpServerMock.createResponseFactory();
+
+    const AppSearchAPI = {
+      shouldBeCalledWith(expectedUrl: string, expectedParams: object) {
+        return {
+          andReturnRedirect() {
+            fetchMock.mockImplementation((url: string, params: object) => {
+              expect(url).toEqual(expectedUrl);
+              expect(params).toEqual(expectedParams);
+
+              return Promise.resolve(
+                new Response('{}', {
+                  url: '/login',
+                })
+              );
+            });
+          },
+          andReturn(response: object) {
+            fetchMock.mockImplementation((url: string, params: object) => {
+              expect(url).toEqual(expectedUrl);
+              expect(params).toEqual(expectedParams);
+
+              return Promise.resolve(new Response(JSON.stringify(response)));
+            });
+          },
+        };
+      },
+    };
+
+    function expectResponseToBe200With(response: object) {
+      expect(mockResponseFactory.ok).toHaveBeenCalledWith(response);
+    }
+
+    async function callThisRoute(
+      request = {
+        headers: {
+          authorization: AUTH_HEADER,
+        },
+        query: {
+          type: TYPE,
+          pageIndex: PAGE_INDEX,
+        },
+      }
+    ) {
+      const [_, handler] = router.get.mock.calls[0];
+
+      const context = {} as jest.Mocked<RequestHandlerContext>;
+      await handler(context, httpServerMock.createKibanaRequest(request), mockResponseFactory);
+    }
+
+    function executeRouteValidation(data: { query: object }) {
+      const [config] = router.get.mock.calls[0];
+      const validate = config.validate as RouteValidatorConfig<{}, {}, {}>;
+      const query = validate.query as ObjectType;
+      query.validate(data.query);
+    }
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      router = mockRouter.create();
+      registerEnginesRoute({
+        router,
+        config: {
+          host: APP_SEARCH_HOST_CONFIG,
+        },
+      });
+    });
+
+    describe('when the underlying App Search API returns a 200', () => {
+      const appSearchEnginesResponse = { name: 'engine1' };
+
+      beforeEach(() => {
+        AppSearchAPI.shouldBeCalledWith(
+          `${APP_SEARCH_HOST_CONFIG}/as/engines/collection?type=${TYPE}&page[current]=${PAGE_INDEX}&page[size]=10`,
+          {
+            headers: { Authorization: AUTH_HEADER },
+          }
+        ).andReturn(appSearchEnginesResponse);
+      });
+
+      it('should return 200 with a list of engines from the App Search API', async () => {
+        await callThisRoute();
+
+        expectResponseToBe200With({
+          body: appSearchEnginesResponse,
```

Comment:
Similar to my comment above - I think there's such thing as too much abstraction when it comes to tests. I know it's repetitive, but it's also useful to be able to open a file and quickly skip to a test and get the expected response in in a single block without having to ctrl+f/skip up and down the file to find what the response looks like. 🤷 

That being said I know others feel differently, so feel free to leave it alone if you strongly disagree.

Reply:
(FWIW I feel the opposite for the AUTH_HEADER const, which we as devs mostly never have to touch or deal with - just forward on - so that makes sense to leave as an abstraction)

Reply:
I put this in a separate method because I didn't think this was very clear to read:

```
expect(mockResponseFactory.ok).toHaveBeenCalledWith({ ...});
```

Meaning, I think it's confusing to look at this and see that we're expecting a response factory "ok" method to be called with a certain body. You kind of miss the intent there ... the intent is that we want the response from the server to be an OK response with a particular body. So it's not really about avoiding the repetition, it's more about expressing intent better.

Reply:
Hmm, ok, that's fair. For me personally as a dev approaching the test as a way to understand the code, it's still useful to have an immediate example of what the body I'm sending _should_ look like, if that makes sense.

Reply:
I misread your comment originally, I thought you were commenting on the `expectResponseToBe200With` method. That's fine I can update this.

Reply:
Updated in 038b792ec5c239ce33cd95bc889359cddf5637b0

Reply:
ohh so sorry, I should have been way more explicit that I was referring to `appSearchEnginesResponse` haha. I think `expectResponseToBe200With` is super understandable/readable :)